### PR TITLE
Added language detection by shebang functionality

### PR DESCRIPTION
**Pull request by mcarmonaa**

### CODE REVIEW

**Comment by smola** [comment]
Let's remove the duplicate templates in the `test_files` directory and always used the ones in `assets`. Do it in a separate PR after this.

File: shebang.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	line = bytes.TrimSpace(line[2:])
+
+	splitted := bytes.Fields(line)
+	if bytes.Contains(splitted[0], []byte("env")) {
+		if len(splitted) > 1 {
+			interpreter = string(splitted[1])
+		}
+
+	} else {
+
```
Comment (alcortesm) [inline shebang.go]: Remove this empty line.

File: shebang.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+func GetLanguageByShebang(data []byte) (lang string, safe bool) {
+	interpreter := getInterpreter(data)
+	lang = OtherLanguage
+	if langs, ok := languagesByInterpreter[interpreter]; ok {
+		lang = langs[0]
+		safe = len(langs) == 1
+	}
+
+	return
```
Comment (mcarmonaa) [inline shebang.go]: Like this function is used as a strateg, you can pass a language to the function that is not a interpreted language so is not in languagesByInterpreter. Then safe must be false to fall to the next strategy.

Comment (alcortesm) [inline shebang.go]: :+1:

**Comment by bzz** [comment]
@mcarmonaa has all the feedback been addressed by now and it's ready for a final review?

**Comment by mcarmonaa** [comment]
@bzz the PR is ready to be merged

**Pull request merged**